unic/utils/src/macros.rs

```diff
@@ -0,0 +1,341 @@
+        queue [ $( $(#[$queue_meta:meta])+ $queue:ident $queue_tt:tt )* ]
+    ) => {
+        __char_property_internal! {
+            $(#[$name_meta])* pub enum $name
+            $(#[$abbr_names_meta])* pub mod $abbr_names
+            $(#[$long_names_meta])* pub mod $long_names
+
+            variant [ $( $(#[$variant_meta])+ $variant ; )* ]
+            abbr [ $( $abbr_variant $abbr ; )* ]
+            long [ $( $long_variant $long ; )* ]
+            display [ $( $display_variant $display ; )* ]
+
+            buffer [ ]
+            queue [ $( $(#[$queue_meta])+ $queue $queue_tt )* ]
+        }
+    };
+
+    // == Final formatting == //
+    (
+        $(#[$name_meta:meta])* pub enum $name:ident
+        $(#[$abbr_names_meta:meta])* pub mod $abbr_names:ident
+        $(#[$long_names_meta:meta])* pub mod $long_names:ident
+
+        variant [ $( $(#[$variant_meta:meta])+ $variant:ident ; )* ]
+        abbr [ $( $abbr_variant:ident $abbr:ident ; )* ]
+        long [ $( $long_variant:ident $long:ident ; )* ]
+        display [ $( $display_variant:ident $display:expr ; )* ]
+
+        buffer [ ]
+        queue [ ]
+    ) => {
+        $(#[$name_meta])*
+        #[allow(bad_style)]
+        #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+        pub enum $name {
+            $( $(#[$variant_meta])+ $variant, )*
+        }
+
+        $(#[$abbr_names_meta])*
+        #[allow(bad_style)]
+        pub mod $abbr_names {
+            $( pub use super::$name::$abbr_variant as $abbr; )*
+        }
+
+        $(#[$long_names_meta])*
+        #[allow(bad_style)]
+        pub mod $long_names {
+            $( pub use super::$name::$long_variant as $long; )*
+        }
+
+        #[allow(bad_style)]
+        #[allow(unreachable_patterns)]
+        impl ::std::str::FromStr for $name {
+            type Err = ();
+            fn from_str(s: &str) -> Result<Self, Self::Err> {
+                match s {
+                    $( stringify!($variant) => Ok($name::$variant), )*
+                    $( stringify!($abbr) => Ok($name::$abbr_variant), )*
+                    $( stringify!($long) => Ok($name::$long_variant), )*
+                    _ => Err(()),
+                }
+            }
+        }
+
+        #[allow(bad_style)]
+        #[allow(unreachable_patterns)]
+        impl ::std::fmt::Display for $name {
+            fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+                match *self {
+                    $( $name::$display_variant => write!(f, "{}", $display), )*
+                    $( $name::$long_variant => write!(f, "{}", stringify!($long).replace('_', " ")), )*
+                    _ => write!(f, "{}", match *self {
+                        $( $name::$abbr_variant => stringify!($abbr), )*
+                        $( $name::$variant => stringify!($variant), )*
+                    })
+                }
+            }
+        }
+
+        #[allow(bad_style)]
+        #[allow(unreachable_patterns)]
```

Comment:
I think you deleted the right one, from one block below, because there used to be three of them. The one on this block (on line 314) is actually needed. The one on line 323 was expected to be deleted, which is now gone.

Uh.... GH BUG!